tests/41end-to-end-keys/01-upload-key.pl

```diff
@@ -162,28 +144,48 @@
       })
    };
 
-push our @EXPORT, qw( matrix_put_e2e_keys );
+push our @EXPORT, qw( matrix_put_e2e_keys matrix_get_e2e_keys );
 
 sub matrix_put_e2e_keys
 {
    # TODO(paul): I don't really know what's parametric about this
    my ( $user, %params ) = @_;
 
    my $device_keys = $params{device_keys} // {};
+   $device_keys->{user_id} = $user->user_id;
```

Comment:
I'm not entirely comfortable with modifying the input like this. maybe copy the hash before adding entries?

Reply:
Fixed.  I think I've done that correctly, but someone who actually knows Perl may want to confirm.

Reply:
looks about right!